File: pid-backend/src/main/java/cz/fit/cvut/pidbackend/Model/Dto/RouteShapeVehicles.java
package cz.fit.cvut.pidbackend.Model.Dto;

import cz.fit.cvut.pidbackend.Model.Route;
import cz.fit.cvut.pidbackend.Model.Shape;
import cz.fit.cvut.pidbackend.Model.Trip;
import cz.fit.cvut.pidbackend.Model.Vehicle;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class RouteShapeVehicles {
    private Route route;
    private Set<Shape> routeShape;
    private Set<Vehicle> vehicles;
}

File: android-app/app/src/main/java/cvut/fit/pidmobapp/model/response/RouteShapeVehicles.kt
package cvut.fit.pidmobapp.model.response

import cvut.fit.pidmobapp.model.Route
import cvut.fit.pidmobapp.model.Shape
import cvut.fit.pidmobapp.model.Vehicle

data class RouteShapeVehicles(
    val route: Route,
    val routeShape: List<Shape>?,
    val vehicles: List<Vehicle>?
)